Cannot export to android if package name contains single quote

### Godot version

3.3.3

### System information

Linux Ubuntu

### Issue description

Cannot export an android build with godot 3.3.3.stable.official because the name "Taila's Jigsaw Adventure" contains a single quote. It worked fine with 3.2.2. Worsk fine with ` bot not with '
`> Task :preBuild UP-TO-DATE
> Task :preDebugBuild UP-TO-DATE
> Task :compileDebugAidl NO-SOURCE
> Task :compileDebugRenderscript NO-SOURCE
> Task :generateDebugBuildConfig UP-TO-DATE
> Task :javaPreCompileDebug UP-TO-DATE
> Task :checkDebugAarMetadata UP-TO-DATE
> Task :generateDebugResValues UP-TO-DATE
> Task :generateDebugResources UP-TO-DATE
> Task :createDebugCompatibleScreenManifests UP-TO-DATE
> Task :extractDeepLinksDebug UP-TO-DATE
> Task :processDebugMainManifest UP-TO-DATE
> Task :processDebugManifest UP-TO-DATE
> Task :processDebugManifestForPackage UP-TO-DATE
> Task :mergeDebugNativeDebugMetadata NO-SOURCE
> Task :mergeDebugShaders UP-TO-DATE
> Task :compileDebugShaders NO-SOURCE
> Task :generateDebugAssets UP-TO-DATE

> Task :mergeDebugResources FAILED
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-de/values-de.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-de/values-de.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-hi/values-hi.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-hi/values-hi.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/07eb4d87e283bf5e3a01bbcaf6469940/jetified-godot-lib.debug/res/values/values.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/07eb4d87e283bf5e3a01bbcaf6469940/jetified-godot-lib.debug/res/values/values.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-pt/values-pt.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-pt/values-pt.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-lt/values-lt.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-lt/values-lt.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-hr/values-hr.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-hr/values-hr.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-lv/values-lv.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-lv/values-lv.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-hu/values-hu.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-hu/values-hu.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-uk/values-uk.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-uk/values-uk.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-in/values-in.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-in/values-in.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-el/values-el.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-el/values-el.xml: string/godot_project_name_string does not contain a valid string resource.
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-zh-rTW/values-zh-rTW.xml: Failed to flatten XML for resource 'godot_project_name_string' with error: Invalid unicode escape sequence in string
"{str}"
/home/puthre/.gradle/caches/transforms-2/files-2.1/ee4998e5c44f8791e55e396202e7f092/core-1.0.0/res/values-zh-rTW/values-zh-rTW.xml: string/godot_project_name_string does not contain a valid string resource.

> Task :mergeDebugAssets UP-TO-DATE

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':mergeDebugResources'.
> Multiple task action failures occurred:
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.
   > A failure occurred while executing com.android.build.gradle.internal.res.ResourceCompilerRunnable
      > Resource compilation failed. Check logs for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
12 actionable tasks: 1 executed, 11 up-to-date

Exit Code: 1`

### Steps to reproduce

Try to export an android project whose package name contains a single quote.

### Minimal reproduction project

_No response_

--- a/platform/android/export/gradle_export_util.cpp
+++ b/platform/android/export/gradle_export_util.cpp
@@ -128,11 +128,26 @@ Error rename_and_store_file_in_gradle_project(void *p_userdata, const String &p_
 	return err;
 }
 
+String _android_xml_escape(const String &p_string) {
+	// Android XML requires strings to be both valid XML (`xml_escape()`) but also
+	// to escape characters which are valid XML but have special meaning in Android XML.
+	// https://developer.android.com/guide/topics/resources/string-resource.html#FormattingAndStyling
+	// Note: Didn't handle U+XXXX unicode chars, could be done if needed.
+	return p_string
+			.replace("@", "\\@")
+			.replace("?", "\\?")
+			.replace("'", "\\'")
+			.replace("\"", "\\\"")
+			.replace("\n", "\\n")
+			.replace("\t", "\\t")
+			.xml_escape(false);
+}
+
 // Creates strings.xml files inside the gradle project for different locales.
 Error _create_project_name_strings_files(const Ref<EditorExportPreset> &p_preset, const String &project_name) {
 	print_verbose("Creating strings resources for supported locales for project " + project_name);
 	// Stores the string into the default values directory.
-	String processed_default_xml_string = vformat(godot_project_name_xml_string, project_name.xml_escape(true));
+	String processed_default_xml_string = vformat(godot_project_name_xml_string, _android_xml_escape(project_name));
 	store_string_at_path("res://android/build/res/values/godot_project_name_string.xml", processed_default_xml_string);
 
 	// Searches the Gradle project res/ directory to find all supported locales
@@ -158,7 +173,7 @@ Error _create_project_name_strings_files(const Ref<EditorExportPreset> &p_preset
 		String locale_directory = "res://android/build/res/" + file + "/godot_project_name_string.xml";
 		if (ProjectSettings::get_singleton()->has_setting(property_name)) {
 			String locale_project_name = ProjectSettings::get_singleton()->get(property_name);
-			String processed_xml_string = vformat(godot_project_name_xml_string, locale_project_name.xml_escape(true));
+			String processed_xml_string = vformat(godot_project_name_xml_string, _android_xml_escape(locale_project_name));
 			print_verbose("Storing project name for locale " + locale + " under " + locale_directory);
 			store_string_at_path(locale_directory, processed_xml_string);
 		} else {

